core/src/main/scala/cromwell/core/path/DefaultPathBuilder.scala

```diff
@@ -12,13 +12,48 @@ import scala.util.Try
   */
 case object DefaultPathBuilder extends PathBuilder {
   override def name = "Default"
-  override def build(pathAsString: String): Try[Path] = Try {
+
+  override def build(pathAsString: String): Try[DefaultPath] = Try {
     val uri = URI.create(UrlEscapers.urlFragmentEscaper().escape(pathAsString))
-    val host = Option(uri.getHost) getOrElse ""
-    val path = Option(uri.getPath) getOrElse ""
     Option(uri.getScheme) match {
-      case Some("file") | None => FileSystems.getDefault.getPath(host, path)
+      case Some("file") | None =>
+
+        if (pathAsString.startsWith("file://")) {
+          // NOTE: Legacy support for old paths generated as URIs by the old .toRealString
+          val host = Option(uri.getHost) getOrElse ""
+          val path = Option(uri.getPath) getOrElse ""
+          DefaultPath(FileSystems.getDefault.getPath(host, path))
+        } else {
+          DefaultPath(FileSystems.getDefault.getPath(pathAsString))
+        }
+
       case _ => throw new RuntimeException(s"Cannot build a local path from $pathAsString")
     }
   }
+
+  /**
+    * Bypasses build()'s URI scheme detection and attempts to directly create a path, providing behavior similar to
+    * java.nio.Paths.get.
+    *
+    * Works even for bad paths like "gs:myfilename.txt", generated by:
+    * centaur 546c001, wf sub_function, call sub, input swappedFile, only when run on JES
```

Comment:
Had to draw the line somewhere. That centaur task is doing some... [interesting](https://github.com/broadinstitute/centaur/blob/546c00180a7b5ec60225f0b215514f7bf9a9b1ff/src/main/resources/standardTestCases/sub_function/sub_function.wdl#L17)... things with `sub()`. The multiple `sub()`s work ok on Local, with previous .toRealString uris, but is creating `gs:myfilename.txt` on JES paths (previously uris).